### ISSUE TITLE
AppTrackingTransparency ios permission prompt on second app launch

### ISSUE THREAD

**Issue by maoryadin** [zoontek/react-native-permissions#720]
### Bug summary

So as I am testing this library, I keep getting the **AppTrackingTransparency** ios permission prompt on the second app launch.
It is weird because I need to close the app and open it for the AppTrackingTransparency permission prompt to appear.

The location permission in the example is prompt at the first launch, as it should be.

### Library version

3.6.1

### Environment info

```shell
System:
    OS: macOS 12.6
    CPU: (8) x64 Intel(R) Core(TM) i5-8279U CPU @ 2.40GHz
    Memory: 2.24 GB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 14.17.1 - ~/.nvm/versions/node/v14.17.1/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 6.14.13 - ~/.nvm/versions/node/v14.17.1/bin/npm
    Watchman: Not Found
  Managers:
    CocoaPods: 1.11.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 16.0, macOS 12.3, tvOS 16.0, watchOS 9.0
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.2 AI-212.5712.43.2112.8609683
    Xcode: 14.0.1/14A400 - /usr/bin/xcodebuild
  Languages:
    Java: 16.0.2 - /Library/Java/JavaVirtualMachines/jdk-16.0.2.jdk/Contents/Home/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1
    react-native: 0.64.3 => 0.64.3
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

### Steps to reproduce

Just run the **requestMultiple** function on an **ios** device:

```
requestMultiple([
    PERMISSIONS.IOS.LOCATION_WHEN_IN_USE,
    PERMISSIONS.IOS.APP_TRACKING_TRANSPARENCY,
])
```

### Reproducible sample code

```js
import {useEffect, useState} from 'react';
import {Platform} from 'react-native';
import {PERMISSIONS, requestMultiple} from 'react-native-permissions';

export const usePermissions = () => {
    const [isPermissionsAsked, setIsPermissionsAsked] = useState(false);

    useEffect(() => {
        console.log('usePermissions useEffect');
        askPermissions();
    }, []);

    const askPermissions = () => {
        if (Platform.OS === 'ios') {
            requestMultiple([
                PERMISSIONS.IOS.LOCATION_WHEN_IN_USE,
                PERMISSIONS.IOS.APP_TRACKING_TRANSPARENCY,
            ])
                .catch(error => {
                    console.log(error);
                }).finally(() => {
                setIsPermissionsAsked(true);
            });
        }

    };

    return [isPermissionsAsked];
};
```

**Comment by zoontek** [comment]
@maoryadin Did you follow this part of the documentation? https://github.com/zoontek/react-native-permissions#how-to-request-app-tracking-transparency-permission-on-ios

**Comment by miladdev85** [comment]
I tried mentioning this problem in the original issue regarding this. My workaround that works and got approved by Apple is to have a async function (delay) for 200ms and then continue with the permission asking flow. 

Works 100% of the time and is not dependent on slow or fast app starts or whether AppState.current is active or not. What I found is that AppState.current could be undefined also and not triggering useEffect when changed.


```
useEffect(() => {
    delay(200).finally(() => permissionStuffOrWhatever());
}, []);
```
